src/pybind/mgr/dashboard/services/ganesha.py

```diff
@@ -118,7 +124,7 @@ def make_rados_url(cls, pool, namespace, obj):
 
     @classmethod
     def get_pool_and_namespace(cls, cluster_id):
-        instances = cls._get_orch_nfs_instances()
+        instances = cls._get_orch_nfs_services()
```

Comment:
`OrchClient.instance().services.list("nfs")`  always returns an empty list because the desired parameter is `service_name`.

Reply:
This is due to this commit https://github.com/ceph/ceph/commit/55adeb93b3f919f202ab79fd93c49fbdf68a7889.  Before that it worked fine. It's better [list()](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/dashboard/services/orchestrator.py#L88) also accepts `service_type`:
 
```
    def list(self, service_type: Optional[str] = None, service_name: Optional[str] = None,) -> List[ServiceDescription]:
        return self.api.describe_service(service_type, service_name)
```

Initially with this PR https://github.com/ceph/ceph/pull/25970, pool and namespace could be fetched even without setting the dashboard `set-ganesha-clusters-rados-pool-namespace`. For maintaining backward compatibility, in rook module `rados_config_location` needs to be set, which was removed by this PR https://github.com/ceph/ceph/pull/33359/files#diff-7321803abaee10bdef98678b409fcf1bL287.

Maybe it is one of the reason for this issue https://github.com/rook/rook/issues/6147.